lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -198,6 +198,36 @@ LogicalResult SubgraphOp::inferReturnTypeComponents(
   return success();
 }
 
+LogicalResult QuantizeOp::inferReturnTypeComponents(
+    MLIRContext *context, ::std::optional<Location> /*location*/,
+    ValueShapeRange operands, DictionaryAttr /*attributes*/,
+    OpaqueProperties /*properties*/, RegionRange /*regions*/,
+    SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes) {
+
+  auto type = llvm::dyn_cast<ShapedType>(operands.front().getType());
+  if (!type || !type.hasStaticShape())
+    return failure();
+
+  auto i8Type = IntegerType::get(context, 8);
+  inferredReturnShapes.push_back(type.clone(i8Type));
```

Comment:
Try and see if you can get the output element type here if not leave a comment.

Reply:
Unfortunately, not possible :(